### DISCUSSION TITLE
Prevent that a Guest-User (member) can change his password and delete his account?

### DISCUSSION THREAD

**Discussion by Tom-Furrer** [fisharebest/webtrees#4913] [General]
I would like to run my webtrees for family members only.

For this people I have created a single "Guest" user and a password and set the access level to "member".

How can I prevent that one of this logged in guests delete this account or change the current password?

Or how do you solve the situation that you only want "granted" people seeing your information without any data change possibility?

**Comment by fisharebest** [comment]
webtrees wasn't really designed for "shared accounts".  I'm not sure any application is.

Why not just create a separate account for each of your family members?

This will give you the advantage of being able to see which members of your family use the site.

**Reply by Tom-Furrer** [reply]
That would be several dozens of new accounts just to "view" the same content with the same restricted access rights :)
In my case that would be too much work for all my relatives.

Probably I have to find the source code entrypoint with the "delete account" possibility .

**Comment by Tom-Furrer** [comment]
I think I have found a quick & dirty way to not show the red "Delete your account"-Button on the "My Account" page anymore.
![image](https://github.com/fisharebest/webtrees/assets/153307907/6da8e587-db67-410d-a614-dc2c5abb8cc4)

For that I set the `show_delete_option` https://github.com/fisharebest/webtrees/blob/cdadf2849bce4d583afc469b550e1dcf6c23dcba/app/Http/RequestHandlers/AccountEdit.php#L95 in Account_edit.php to a fixed 0 (Zero). The current line number is 95.

The changed line should look like this:
`'show_delete_option'   => '0',`

**Comment by Tom-Furrer** [comment]
If you don't want to show the submenu "My Account" in the main menu "My pages" 
and the submenu "Customize this page" on "My page" (shown as "My pages") at all,
you can comment out this two lines in app/Module/ModuleThemeTrait.php,
first  #L372 
https://github.com/fisharebest/webtrees/blob/cdadf2849bce4d583afc469b550e1dcf6c23dcba/app/Module/ModuleThemeTrait.php#L372

and second #L374
https://github.com/fisharebest/webtrees/blob/cdadf2849bce4d583afc469b550e1dcf6c23dcba/app/Module/ModuleThemeTrait.php#L374

looking like this:
![image](https://github.com/fisharebest/webtrees/assets/153307907/bca8ceb6-e685-40fb-9f83-38328504617a)

Please remember:
This is just a quick & dirty way. 

It would be much more elegant to have an official option to enable/disable this entries.
As this may be far too specific as general change, it could give an idea how to create an indidivual workaround.

Don't forget that you have to do such changes again with every new webtrees update you apply!
